### DISCUSSION TITLE
Error: `useUIState` must be used inside an <AI> provider.

### DISCUSSION THREAD

**Discussion by KolyaDychkevych** [vercel/next.js#67243] [Help]
### Summary

in order for the hooks ( { useActions, useUIState } from "ai/rsc"; ) to work, it was necessary to wrap the RootLayout in AI provider and everything works ok

but when my next project structure has folder SRC and all the files are there(pages, etc), it throws an error.
Error: `useUIState` must be used inside an <AI> provider.
maybe because my layout with the AI ​​provider remained in the app folder and how can I change it?

### Additional information

```js
export const AI = createAI<ServerMessage[], ClientMessage[]>({
  actions: {},
  initialAIState: [],
  initialUIState: [],
});

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={inter.className}>
      <AI>
        {children}
      </AI>
      </body>
    </html>
  );
}
```

**Comment by brandon-pereira** [comment]
For me, I copied the boilerplate code in the docs, but the API has changed slightly.

1. I had to import `AI` that is exported from `createAI` on the backend. This is the provider.
2. The frontend immediately called `useUIState` but that is incorrect because the provider was missing, so I had to create a parent component which wrapped the provider and then rendered by sub-component which called the hook `useUIState`.

Hope it helps! Wasn't hard just lacking documentation.
